src/dune_rules/workspace.ml

```diff
@@ -6,10 +6,40 @@ open Dune_lang.Decoder
    simplicity *)
 let syntax = Stanza.syntax
 
+let binaries_in_any e =
```

Comment:
Note that the issue is not specific to `Any`, it also triggers when a profile is specified.

``` 
(env
 (dev
   (binaries (some-path as node))))
```